scalableminds/webknossos
# Handle slow save requests better

- warn the user when there are unsaved changes older than X seconds (for testing, X is set to 5 seconds; for production, I'd suggest to set it to 2 minutes)
- send a warning to airbrake when a save request takes longer than 1 minute (set to 5 s for testing)
- send a a warning to airbrake when there was a conflict state (status code 409)

![image](https://user-images.githubusercontent.com/2486553/81068456-b04ee380-8ee0-11ea-911b-3b1ddeb846dd.png)


### URL of deployed dev instance (used for testing):
- https://debugsavetimeout.webknossos.xyz

### Steps to test:
- modify a tracing and a warning at the save button should appear after 5 seconds. the back-end has a hardcoded sleep which should make the save request extra slow

### Issues:
- solves front-end part of #4594

------
